solutions/platinum/cf-1254D.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: cf-1254D
+source: CF
+title: Tree Queries
+author: Rameez Parwez
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/71594)
```

Comment:
can you write an editorial for this problem?

Reply:
Well, that is the issue, I can't process my thoughts clearly on this. Plus I have seen in a few internal solutions that there was no explanation only the official editorial link was provided that's why I adopted it.

Reply:
The official editorial doesn't use HLD and has a completely different time complexity than yours.

Reply:
Since the topic was about HLD I wanted the solution using that. 
I mentioned the official editorial link because I have seen in many solutions that it is always mentioned at the top.

Reply:
Yes, using HLD is good. However, I think only having code doesn't add much value. I still think it's going to be hard for readers to understand your code without having an explanation.

Reply:
I will try to add some explanation to it.

Reply:
Added some explanation though I am not sure whether you will like it or not.